pyfr/backends/openmp/kernels/batch-gemm.mako

```diff
@@ -4,19 +4,33 @@
 struct kargs
 {
     void (*exec)(void *, const fpdtype_t *, fpdtype_t *);
-    void *blockk;
     int nblocks;
     const fpdtype_t *b;
     int bblocksz;
     fpdtype_t *c;
     int cblocksz;
+    void *blockk[];
 };
 
 void batch_gemm(const struct kargs *restrict args)
 {
     #pragma omp parallel for ${schedule}
     for (int ib = 0; ib < args->nblocks; ib++)
-        args->exec(args->blockk,
+    {
+      % for i in range(nfac):
+        % if loop.first and loop.last:
+        args->exec(args->blockk[0],
                    args->b + ib*args->bblocksz,
                    args->c + ib*args->cblocksz);
+        % elif loop.first:
+        fpdtype_t buf0[${blocksz}]${';' if nfac == 2 else f', buf1[{blocksz}];'}
```

Comment:
We will need to use `alisgnas` here from `stdalign.h` and may need to compile in C11 mode.

Reply:
Also these tests might need some more work.  I'd be inclined to move these definitions outside of the `% for` loop.